File: app/src/main/java/com/abdelmeged/ahmed/networkitotest/java/ArchActivity.java
package com.abdelmeged.ahmed.networkitotest.java;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;

import com.abdelmeged.ahmed.networkito.ConnectionType;
import com.abdelmeged.ahmed.networkito.ConnectivityChangeListener;
import com.abdelmeged.ahmed.networkito.Networkito;
import com.abdelmeged.ahmed.networkitotest.R;


public class ArchActivity extends AppCompatActivity implements ConnectivityChangeListener {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        new Networkito(this, this, this);
    }

    @Override
    public void onInternetConnected(ConnectionType connectionType) {
        Toast.makeText(this, "connected: " + connectionType.name(), Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onInternetDisconnected() {
        Toast.makeText(this, "disconnected", Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/abdelmeged/ahmed/networkitotest/kotlin/ArchActivity.kt
package com.abdelmeged.ahmed.networkitotest.kotlin

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.abdelmeged.ahmed.networkito.ConnectionType
import com.abdelmeged.ahmed.networkito.ConnectivityChangeListener
import com.abdelmeged.ahmed.networkito.Networkito
import com.abdelmeged.ahmed.networkitotest.R
import org.jetbrains.anko.toast

class ArchActivity : AppCompatActivity(), ConnectivityChangeListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Networkito(this, this, this)
    }

    override fun onInternetConnected(connectionType: ConnectionType?) {
        toast("connected: " + connectionType!!.name)
    }

    override fun onInternetDisconnected() {
        toast("disconnected")
    }

}
